File: dyn/tor/torcheck.py
def application(environ, start_response):
    #response_body = "Not using Tor, ip: " + environ['REMOTE_ADDR'] # Just assume
    response_body = "#tor{color:#d75f00 !important}"
    with open("/var/cache/dyn/exits.txt") as f:
        for l in f:
            if environ['REMOTE_ADDR'] == l.strip('\n'):
                #response_body = "Congrats for using Tor, ip: " \
                #        + environ['REMOTE_ADDR']
                response_body = "#tor{color:#008700 !important}"
                break

    response_headers = [
        ('Content-Type', 'text/css'),
    ]

    start_response('200 OK', response_headers)

    yield response_body.encode('utf-8')
    yield b'\n'
#    yield ('\n'.join(['%s: %s' % (key, value) for key, value in sorted(environ.items())])).encode('utf-8')

File: _dyn/tor/torcheck.c
/* CGI to match if client connect as tor */
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

#define TEXT "/var/cache/dyn/exits.txt"
#define HOST "pickfireywcq2wf2.onion"

int
main(int argc, char *argv[])
{
	FILE *fp;
	char *ip, buf[16];
	int tor = 0;

	if (strstr(HOST, getenv("SERVER_NAME")) != NULL)
		tor = 1;
	else {
		ip = strncat(getenv("REMOTE_ADDR"), "\n", 15);

		if (!(fp = fopen(TEXT, "r")))
			return 1;
		while (fgets(buf, 16, fp))
			if (!strncmp(buf, ip, 15)) {
				tor = 1;
				break;
			}
		fclose(fp);
	}

	printf("Content-type: text/css\r\n\r\n#tor{color:%s !important}",
		tor ? "#008700" : "#df0000");
	return 0;
}
